src/core/types/instances.jl

```diff
@@ -49,22 +49,22 @@ end
 
 # A container class that wraps the dimension dictionary when passed to run_timestep()
 # and init(), so we can safely implement Base.getproperty(), allowing `d.regions` etc.
-# All values in the dictionary are vectors of Ints, except the `:time` value, which is a
+# All values in the named tuple are vectors of Ints, except the `:time` value, which is a
 # vector of AbstractTimesteps, so that `d.time` returns values that can be used for indexing
 # into timestep arrays.
-struct DimValueDict <: MimiStruct
-    dict::Dict{Symbol, Union{Vector{Int}, Vector{AbstractTimestep}}}
+struct DimValueNamedTuple <: MimiStruct
```

Comment:
So this won't solve the problem, because the new type is still not parameterized properly. One way to do this would be to write
```
struct DimValueNamedTuple{T<:NamedTuple} <: MimiStruct
  nt::T
end
```
But I think we can just remove this type entirely and simply pass a named tuple directly to all the user provided functions.